nifi-extension-bundles/nifi-aws-bundle/nifi-aws-abstract-processors/src/main/java/org/apache/nifi/processors/aws/s3/AbstractS3Processor.java

```diff
@@ -207,230 +147,120 @@ public abstract class AbstractS3Processor extends AbstractAWSCredentialsProvider
             .defaultValue("false")
             .build();
 
-    /**
-     * Create client using credentials provider. This is the preferred way for creating clients
-     */
-    @Override
-    protected AmazonS3Client createClient(final ProcessContext context, final AWSCredentialsProvider credentialsProvider, final Region region,
-                                          final ClientConfiguration config, final AwsClientBuilder.EndpointConfiguration endpointConfiguration) {
-        getLogger().info("Creating client with credentials provider");
-        initializeSignerOverride(context, config);
-        AmazonS3EncryptionService encryptionService = context.getProperty(ENCRYPTION_SERVICE).asControllerService(AmazonS3EncryptionService.class);
-
-        final Consumer<AmazonS3Builder<?, ?>> clientBuilder = builder -> {
-            if (endpointConfiguration == null) {
-                builder.withRegion(region.getName());
-            } else {
-                builder.withEndpointConfiguration(endpointConfiguration);
-            }
-            builder.withClientConfiguration(config);
-            builder.withCredentials(credentialsProvider);
-
-            final Boolean useChunkedEncoding = context.getProperty(USE_CHUNKED_ENCODING).asBoolean();
-            if (useChunkedEncoding == Boolean.FALSE) {
-                builder.disableChunkedEncoding();
-            }
-
-            final Boolean usePathStyleAccess = context.getProperty(USE_PATH_STYLE_ACCESS).asBoolean();
-            final boolean endpointOverrideSet = !StringUtils.trimToEmpty(context.getProperty(ENDPOINT_OVERRIDE).evaluateAttributeExpressions().getValue()).isEmpty();
-            if (usePathStyleAccess == Boolean.TRUE || endpointOverrideSet) {
-                builder.withPathStyleAccessEnabled(true);
-            }
-        };
-
-        AmazonS3 s3Client = null;
-        if (encryptionService != null) {
-            s3Client = encryptionService.createEncryptionClient(clientBuilder);
-        }
-        if (s3Client == null) {
-            final AmazonS3ClientBuilder builder = AmazonS3Client.builder();
-            clientBuilder.accept(builder);
-            s3Client = builder.build();
-        }
-
-        return (AmazonS3Client) s3Client;
-    }
-
+    // maps AWS SDK v1 CannedAccessControlList to v2 ObjectCannedACL
+    private static final Map<String, ObjectCannedACL> CANNED_ACL_MAPPING = Map.of(
+            "Private", ObjectCannedACL.PRIVATE,
+            "PublicRead", ObjectCannedACL.PUBLIC_READ,
+            "PublicReadWrite", ObjectCannedACL.PUBLIC_READ_WRITE,
+            "AuthenticatedRead", ObjectCannedACL.AUTHENTICATED_READ,
+            "LogDeliveryWrite", ObjectCannedACL.UNKNOWN_TO_SDK_VERSION,
```

Comment:
When `CANNED_ACL` is set to `LogDeliveryWrite`, we now map it to `ObjectCannedACL.UNKNOWN_TO_SDK_VERSION`. In the SDK v2 generated models, that would serialize to an empty value, so the `x-amz-acl` header would never be sent I believe.

Previously we sent `LogDeliveryWrite`, which is required when configuring the bucket to receive S3 server access logs. With this change, the request reverts to the default (effectively private) and S3 would reject the logging configuration.

We need to send log-delivery-write explicitly instead of `UNKNOWN_TO_SDK_VERSION`.

```suggestion
            "LogDeliveryWrite", ObjectCannedACL.LOG_DELIVERY_WRITE,
```